### PR TITLE
KAFKA-4112: Remove alpha quality label from Kafka Streams in docs

### PR DESCRIPTION
**Pull request by dguy**
Rephrase 'alpha quality' wording in Streams section of api.html.
Couple of other minor fixes in streams.html

### CODE REVIEW

**Comment by dguy** [comment]
@miguno @ijuma please review

File: docs/api.html [hunk 1]
```diff
@@ -169,7 +169,7 @@
 <h3><a id="streamsapi" href="#streamsapi">2.3 Streams API</a></h3>
 
 As of the 0.10.0 release we have added a new client library named <b>Kafka Streams</b> to let users implement their stream processing
-applications with data stored in Kafka topics. Kafka Streams is considered alpha quality and its public APIs are likely to change in
+applications with data stored in Kafka topics. Kafka Streams is a new library and its public APIs are likely to change in
```
Comment (miguno) [inline docs/api.html]: The updated section also has the following paragraph at the very end:

> Examples showing how to use this library are given in the javadocs (note those classes annotated with @InterfaceStability.Unstable, indicating their public APIs may change without backward-compatibility in future releases). 

So perhaps we should reword/merge the section to sth like the following (including some tweaks to explaining what Kafka Streams is, and bumping the version string to the latest release):

> As of the 0.10.0 release we have added a stream processing engine to Apache Kafka called **Kafka Streams**, which is a client library that lets users implement their own stream processing applications for data stored in Kafka topics. You can use Kafka Streams from within your Java applications by adding a dependency on the kafka-streams jar using the following maven co-ordinates:
> 
> ```
>     <dependency>
>         <groupId>org.apache.kafka</groupId>
>         <artifactId>kafka-streams</artifactId>
>         
>         <version>0.10.0.1</version>
>     </dependency>
> ```
> 
> Examples showing how to use this library are given in the [Kafka Streams javadocs](http://kafka.apache.org/0100/javadoc/index.html?org/apache/kafka/streams/KafkaStreams.html).
> 
> Please note that Kafka Streams is a new component of Kafka, and its public APIs may change in future releases.  We use the `@InterfaceStability.Unstable` annotation to denote classes whose APIs may change without backward-compatibility in future releases.

File: docs/streams.html [hunk 2]
```diff
@@ -260,7 +260,7 @@
     KStreamBuilder builder = new KStreamBuilder();
 
     KStream<String, GenericRecord> source1 = builder.stream("topic1", "topic2");
-    KTable<String, GenericRecord> source2 = builder.table("topic3");
+    KTable<String, GenericRecord> source2 = builder.table("topic3", "stateStoreName");
```
Comment (miguno) [inline docs/streams.html:263]: Nice catches!

**Comment by dguy** [comment]
Updated. thanks @miguno

**Comment by miguno** [comment]
LGTM! Thanks @dguy

File: docs/api.html [hunk 3]
```diff
@@ -168,19 +168,20 @@
...
 	    &lt;artifactId&gt;kafka-streams&lt;/artifactId&gt;
-	    &lt;version&gt;0.10.0.0&lt;/version&gt;
+	    &lt;version&gt;0.10.0.1&lt;/version&gt;
 	&lt;/dependency&gt;
 </pre>
 
 Examples showing how to use this library are given in the
-<a href="http://kafka.apache.org/0100/javadoc/index.html?org/apache/kafka/streams/KafkaStreams.html" title="Kafka 0.10.0 Javadoc">javadocs</a> (note those classes annotated with <b>@InterfaceStability.Unstable</b>, indicating their public APIs may change without backward-compatibility in future releases).
\ No newline at end of file
+<a href="http://kafka.apache.org/0100/javadoc/index.html?org/apache/kafka/streams/KafkaStreams.html" title="Kafka 0.10.0 Javadoc">javadocs</a>.
```
Comment (ijuma) [inline docs/api.html]: Do we want to mention the Streams section in the docs that includes a lot of useful information as well?

**Comment by ijuma** [comment]
It would be good to have a JIRA for this so that it appears in the release notes.

**Comment by dguy** [comment]
Thanks @ijuma. done

**Comment by ijuma** [comment]
LGTM

**Comment by guozhangwang** [comment]
LGTM, merged to trunk and cherry picked to 0.10.0.

**Pull request closed without merging**